plugins/inputs/win_eventlog/util.go

```diff
@@ -97,12 +98,10 @@ func UnrollXMLFields(data []byte, fieldsUsage map[string]int, separator string)
 	for {
 		var node xmlnode
 		err := dec.Decode(&node)
-		if err == io.EOF {
-			break
-		}
-		if err != nil {
+		if errors.Is(err, io.EOF) || err != nil {
```

Comment:
I think this is the same
```suggestion
		if err != nil {
```

Reply:
Good catch!